src/common.rs

```diff
@@ -1572,10 +1572,49 @@ pub fn display_access_key_list(access_keys: &[near_primitives::views::AccessKeyI
     table.printstd();
 }
 
+/// sort network names based on account_id if provided
+/// if account_id is not provided, sort based on network name
+/// this is to make sure that the default network is always the same
+/// otherwise, sort the network names which maps to the same account_id to the top of the list,
+/// and then sort the rest of the network names alphabetically
 pub fn input_network_name(
     config: &crate::config::Config,
+    account_id: Option<crate::types::account_id::AccountId>,
 ) -> color_eyre::eyre::Result<Option<String>> {
-    let variants = config.network_connection.keys().collect::<Vec<_>>();
+    let variants = if let Some(account_id) = account_id {
+        let account_id =
+            <crate::types::account_id::AccountId as AsRef<str>>::as_ref(&account_id).to_string();
+        let vec_account_ids = account_id.split(".").collect::<Vec<_>>();
+        let mut variants = config.network_connection.keys().collect::<Vec<_>>();
+        variants.sort_by(|a, b| -> std::cmp::Ordering {
+            let a_account_id = config
+                .network_connection
+                .get(a.to_owned())
+                .unwrap()
+                .network_name
+                .clone();
+            let b_account_id = config
+                .network_connection
+                .get(b.to_owned())
+                .unwrap()
+                .network_name
+                .clone();
+
+            if a_account_id.contains(vec_account_ids[1]) {
+                std::cmp::Ordering::Less
+            } else if b_account_id.contains(vec_account_ids[1]) {
+                std::cmp::Ordering::Greater
+            } else {
+                a.cmp(b)
+            }
+        });
```

Comment:
* [ ] We don't need to sort the list, just lift up more relevant options
* [ ] Don't use `unwrap` (in this specific case it could have been avoided by using `iter()` instead of `keys()`)
* [ ] Don't use `[1]` as it may crash if account id does not have `.` (e.g. implicit account name)
* [ ] We only need the last prefix of the specified account id